CMake/External_GDAL.cmake

```diff
@@ -67,9 +67,15 @@ else()
     #
     # Also set '--with-local=/usr' to force selection of /usr/lib/libiconv over e.g. /opt/local/lib
     # from macports.  GDAL's '--with-libiconv-prefix' option looks like it should handle
-    # this but in fact seems to do nothing.
-    #
-    set(_GDAL_ARGS_APPLE --without-libtool --with-netcdf=no --with-curl=no --with-local=/usr)
+    # this but in fact seems to do nothing. 
+
+    if(fletch_ENABLE_libkml)
+      #If we're building libkml on apple, then use it. 
+      list(APPEND _GDAL_DEPENDS libkml)
+      set(_GDAL_ARGS_APPLE --with-libkml=${KML_DIR} --without-libtool --with-netcdf=no --with-curl=no --with-local=/usr)
```

Comment:
I think `KML_DIR` here needs to be `KML_ROOT` instead (or just `fletch_ROOT`) because `KML_DIR` points to the CMake exports file, not the install root. 

This change almost works.  It finds the headers, but not the library, and marks libkml as not found and then I get the same error as before.  I only have "libkml.a", maybe it's looking for the dynamic library?